common/src/paused_program_storage.rs

```diff
@@ -0,0 +1,94 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use crate::storage::MapStorage;
+use core::fmt::Debug;
+
+#[derive(Clone, Debug, PartialEq, Eq, Decode, Encode, TypeInfo)]
+#[codec(crate = codec)]
+#[scale_info(crate = scale_info)]
+struct Item {
+    allocations: BTreeSet<WasmPage>,
+    memory_pages: BTreeMap<GearPage, PageBuf>,
+    code_hash: H256,
+    state: ProgramState,
```

Comment:
do we need state here?

Reply:
Yep, in other case we would store it alongside with the hash

Reply:
discussed: uninitialized programs will be terminated instead of being paused.